nucliadb_node/src/settings/mod.rs

```diff
@@ -0,0 +1,302 @@
+    max_shards_per_node: usize,
+
+    // Index node self data
+    #[builder(default = "\"host_key\".into()", setter(name = "host_key_path"))]
+    node_id_path: PathBuf,
+    #[builder(default = "reliable_lookup_host(\"localhost\").ip()", setter(custom))]
+    public_ip: IpAddr,
+
+    // Errors
+    #[builder(default = "String::new()")]
+    sentry_url: String,
+    #[builder(default = "SENTRY_DEV", setter(custom))]
+    sentry_env: &'static str,
+
+    // Logs and traces
+    #[builder(default = "parse_log_levels(\"nucliadb_node=WARN,nucliadb_cluster=WARN\")")]
+    log_levels: Vec<(String, Level)>,
+    #[rustfmt::skip]
+    #[builder(
+        default = "parse_log_levels(\"nucliadb_node=INFO,nucliadb_cluster=INFO,nucliadb_core=INFO\")",
+        setter(skip)
+    )]
+    span_levels: Vec<(String, Level)>,
+
+    // Telemetry
+    #[builder(default = "false", setter(custom))]
+    jaeger_enabled: bool,
+    #[builder(default = "\"localhost\".into()")]
+    jaeger_agent_host: String,
+    #[builder(default = "6831")]
+    jaeger_agent_port: u16,
+
+    #[builder(default = "reliable_lookup_host(\"localhost:40102\")", setter(custom))]
+    reader_listen_address: SocketAddr,
+    #[builder(default = "reliable_lookup_host(\"localhost:40101\")", setter(custom))]
+    writer_listen_address: SocketAddr,
+
+    // Cluster
+    #[builder(default = "40100")]
+    chitchat_port: u16,
+    #[builder(default = "Vec::new()")]
+    seed_nodes: Vec<String>,
+    #[builder(default = "Duration::from_millis(500)")]
+    cluster_liveliness_update_interval: Duration,
+
+    #[builder(default = "Duration::from_secs(5)")]
+    shutdown_delay: Duration,
+
+    #[builder(default = "3030")]
+    metrics_port: u16,
+}
+
+impl Settings {
+    pub fn builder() -> SettingsBuilder {
+        SettingsBuilder::default()
+    }
+
+    /// Path to main directory where all index node data is stored
+    pub fn data_path(&self) -> PathBuf {
+        self.data_path.clone()
+    }
+
+    /// When shard lazy loading is enabled, reader and writer will try to load a
+    /// shard before using it. Otherwise, they'll load all shards at startup
+    pub fn lazy_loading(&self) -> bool {
+        self.lazy_loading
+    }
+
+    /// Maximum number of shards an index node will store
+    pub fn max_shards_per_node(&self) -> usize {
+        self.max_shards_per_node
+    }
+
+    // TODO: rename to `node_id_path` or similar
+    /// Path to index node UUID file
+    pub fn host_key_path(&self) -> PathBuf {
+        self.node_id_path.clone()
+    }
+
+    /// Host public IP
+    pub fn public_ip(&self) -> IpAddr {
+        self.public_ip
+    }
+
+    pub fn sentry_url(&self) -> String {
+        self.sentry_url.clone()
+    }
+
+    /// Sentry environment to report errors
+    pub fn sentry_env(&self) -> &'static str {
+        self.sentry_env
+    }
+
+    /// Log levels. Every element is a crate-level pair
+    pub fn log_levels(&self) -> Vec<(String, Level)> {
```

Comment:
Is a better practice to return `&[(String, Level)]` and let the caller decide how and when the allocation should be done.